File: makeArray2.py
# for while 禁止

i = 1
x = []

def makeArray2():
    global i
    global x
    if i>30:
        return print(x)
    if i%3==0 or i%5==0 or i%7==0 or i%11==0:
        x.append(i)
    i += 1
    makeArray2()
        
            
makeArray2()

File: makeArray2.js
const makeArray2 = function() {
    let x = [];
    let i = 0;
    (function push_i() {
        i++;
        if(i>30) { return; }
        if(i%3===0 || i%5===0 || i%7===0 || i%11===0) { x.push(i); } 
        push_i()       
    })();
    return x;
}

console.log(makeArray2());

